corehq/messaging/scheduling/views.py

```diff
@@ -280,10 +316,22 @@ def page_context(self):
         return {
             'basic_info_form': self.basic_info_form,
             'criteria_form': self.criteria_form,
+            'schedule_form': self.schedule_form,
         }
 
+    @cached_property
+    def schedule_form(self):
+        if self.request.method == 'POST':
+            return ConditionalAlertScheduleForm(self.domain, self.schedule, self.rule, self.request.POST)
+
+        return ConditionalAlertScheduleForm(self.domain, self.schedule, self.rule)
+
+    @property
+    def schedule(self):
+        return None
+
     @property
-    def initial_rule(self):
+    def rule(self):
         return None
```

Comment:
Nit: could these be simple class attributes?
```py
schedule = None
rule = None
```

Reply:
i kind of liked keeping it with a property definition to help reinforce that it should be overridden in the subclass, and also because since they don't have a setter it's not possible to accidentally give them a value and get things mixed up in the base class